src/features/dotnetTest.ts

```diff
@@ -35,41 +33,124 @@ export function registerDotNetTestDebugCommand(server: OmniSharpServer): vscode.
-    getTestOutputChannel().appendLine('Running test ' + testMethod + '...');
-    serverUtils
-        .runDotNetTest(server, { FileName: fileName, MethodName: testMethod, TestFrameworkName: testFrameworkName })
-        .then(
-        response => {
+    const output = getTestOutputChannel();
+
+    output.show();
+    output.appendLine(`Running test ${testMethod}...`);
+
+    const disposable = server.onTestMessage(e => {
+        output.appendLine(e.Message);
+    });
+
+    const request: protocol.V2.RunTestRequest = {
+        Filename: fileName,
+        MethodName: testMethod,
+        TestFrameworkName: testFrameworkName
+    };
+
+    serverUtils.runTest(server, request)
+        .then(response => {
             if (response.Pass) {
-                getTestOutputChannel().appendLine('Test passed \n');
+                output.appendLine('Test passed \n');
             }
             else {
-                getTestOutputChannel().appendLine('Test failed \n');
+                output.appendLine('Test failed \n');
             }
+
+            disposable.dispose();
         },
         reason => {
             vscode.window.showErrorMessage(`Failed to run test because ${reason}.`);
+            disposable.dispose();
         });
 }
 
+function getLaunchConfigurationForAttach(server: OmniSharpServer, fileName: string, testMethod: string, testFrameworkName: string): Promise<any> {
+    const request: protocol.V2.GetTestStartInfoRequest = {
+        Filename: fileName,
+        MethodName: testMethod,
+        TestFrameworkName: testFrameworkName
+    };
+
+    return serverUtils.debugTestStart(server, request)
+        .then(response => {
+            return {
+                name: ".NET Test Attach",
+                type: "coreclr",
+                request: "attach",
+                processId: response.ProcessId
+            };
+        });
+}
+
+function getLaunchConfigurationForLaunch(server: OmniSharpServer, fileName: string, testMethod: string, testFrameworkName: string): Promise<any> {
+    const request: protocol.V2.GetTestStartInfoRequest = {
+        Filename: fileName,
+        MethodName: testMethod,
+        TestFrameworkName: testFrameworkName
+    };
+
+    return serverUtils.getTestStartInfo(server, request)
+        .then(response => {
+            let args = response.Argument.split(' ');
+
+            // Ensure that quoted args are unquoted.
+            args = args.map(arg => {
+                if (arg.startsWith('"') && arg.endsWith('"')) {
+                    return arg.substring(1, arg.length - 1);
+                }
+                else {
+                    return arg;
+                }
+            });
+
+            return {
+                name: ".NET Test Launch",
+                type: "coreclr",
+                request: "launch",
+                program: response.Executable,
+                args: args,
+                cwd: response.WorkingDirectory,
+                stopAtEntry: false
```

Comment:
Should we be including the environment variables also? Or is that collection always empty?